src/components/common/form/Form.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { StyleSheet, View } from "react-native";
+import { useForm } from "react-hook-form";
+import FormInput from "./FormInput";
+import Button, { BUTTON_TYPES } from "../Button";
+
+const Form = ({ inputs, defaultValues, onSubmit, isLoading, buttonText }) => {
+	const { control, handleSubmit } = useForm({
+		defaultValues: defaultValues
+	});
+
+	return (
+		<View style={styles.formContainer}>
+			{inputs.map((item, index) => (
+				<FormInput
+					key={index}
+					element={item}
+					control={control}
+					rules={item?.rules}
+				/>
+			))}
+			<Button
+				text={buttonText}
+				onPress={handleSubmit(onSubmit)}
+				style={styles.loginButton}
+				textStyle={styles.buttonText}
+				type={BUTTON_TYPES.PRIMARY}
+				isLoading={isLoading}
```

Comment:
when it's loading, this shouldn't load, but it should be disabled right? Maybe we add a disabled state to the button